server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -136,7 +144,8 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
             .flatMap(ignored -> readFromCache(blobId)
                 .switchIfEmpty(readBytesFromBackend(bucketName, blobId)
                     .filter(this::isAbleToCache)
-                    .flatMap(bytes -> saveInCache(blobId, bytes).then(Mono.just(bytes)))))
+                    .flatMap(bytes -> saveInCache(blobId, bytes).then(Mono.just(bytes))))
+                .map(this::increaseCacheMetricCount))
```

Comment:
We should rely on doOnNext and maybe relax our assertions (with awaits) if needed